### PR TITLE
SMTP_OPENSSL_VERIFY_MODE should not be set to 'peer' unless there is a client certificate

### PR DESCRIPTION
**Pull request by progval**
Otherwise, this gives the following error:

```
OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 state=error: certificate verify failed (unable to get local issuer certificate)
```

PS: I don't know how to write Ruby code, please check the syntax is ok

### CODE REVIEW

File: lib/tasks/mastodon.rake [hunk 1]
```diff
@@ -256,10 +256,11 @@ namespace :mastodon do
             q.modify :strip
           end
 
-          env['SMTP_OPENSSL_VERIFY_MODE'] = prompt.ask('SMTP OpenSSL verify mode:') do |q|
-            q.required
-            q.default 'peer'
-            q.modify :strip
+          if prompt.yes?('Do you want to use a client certificate to authenticate to the server?', default: false)
+            env['SMTP_OPENSSL_VERIFY_MODE'] = 'peer'
+            prompt.ask('Path to the client certificate:') do |q|
```
Comment (nightpool) [inline lib/tasks/mastodon.rake]: This client certificate never gets put into the environment anywhere?

Comment (ProgVal) [reply lib/tasks/mastodon.rake]: Oops, indeed. I'm going to fix it

Comment (ProgVal) [reply lib/tasks/mastodon.rake]: Hmm, actually, Mastodon does not support client certificates. Should we nuke `SMTP_OPENSSL_VERIFY_MODE`?

Comment (akihikodaki) [reply lib/tasks/mastodon.rake]: Didn't you notice this because you are going to use this feature? Or were you just reviewing the code?

Comment (Gargron) [reply lib/tasks/mastodon.rake]: I also noticed that mastodon:setup-generated SMTP config failed for me until I removed `SMTP_OPENSSL_VERIFY_MODE` from it, when setting up a new instance.

Comment (ProgVal) [reply lib/tasks/mastodon.rake]: @akihikodaki I noticed this because someone on IRC had that issue.

**Comment by ineffyble** [comment]
@ProgVal Could you merge master from tootsuite into your branch, to fix the CI fails on this PR?

**Comment by ProgVal** [comment]
@ineffyble Done.

File: lib/tasks/mastodon.rake (hunk 1 above)
Comment (akihikodaki) [reply lib/tasks/mastodon.rake]: Sorry for late answer:
> Hmm, actually, Mastodon does not support client certificates. Should we nuke `SMTP_OPENSSL_VERIFY_MODE`?

I think so.

**Pull request closed without merging**